lib/pbench/server/api/resources/api_key.py

```diff
@@ -32,6 +32,9 @@ def __init__(self, config: PbenchServerConfig):
                 query_schema=Schema(
                     Parameter("label", ParamType.STRING, required=False),
                 ),
+                uri_schema=Schema(
+                    Parameter("key", ParamType.STRING, required=False),
+                ),
```

Comment:
This is unfortunate. The `ApiParams` won't have a `uri` attribute if there was no `uri_schema` for the operation, and the API implementation normally has no access to the original `_dispatch` dict.

*However*, in order to collect the original "raw" API parameters for `GET /datasets` pagination, I recently added a `raw_params` field to the `context`: so instead of defining a parameter in the schema just to fail if it's used, you should be able check for the parameter like this:

```python
    if context["raw_params"].uri:
        raise APIAbort(HTTPStatus.BAD_REQUEST, "the path you are taking will lead to certain doom")
```

(Although your message is more "professional" and possibly more useful. 😆 )

You could use `if "key" in context["raw_params"].uri` ... but we don't expect *any* URI parameters, so `uri` ought to be `{}`, which is falsey.